### PR TITLE
Add contributing instructions

### PR DESCRIPTION
**Pull request by mattfrankjames**
Addresses this [issue](https://github.com/ember-components/ember-component-patterns/issues/4).
- Adds a basic contributing document to the repo.

### CODE REVIEW

**Review by hergaiety — COMMENTED** [review]
Looking superb! Thanks for tackling this. Few change suggestions but only the typo is required.

Also wanted to suggest that the `## Guidelines for contribution` include mention of how to test against these things. Even if we simply say `npm test` and `npm run lint:js` are available and the rest has to be done via a pull request firing off the other things in CI. Otherwise contributors may feel like there are steps they have to take but they don't know how to perform them.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+Contributions of all types are welcome. Whether filing issues, testing components against screen readers, submitting examples of anti-patterns or contributing to the code base, all PRs are welcome. It is preferred that each PR be linked to a documented issue, though this is not strictly neccessary if there is no issue for a given PR.
+
+## Getting started
+
+In order to contribute to this repo, please follow the following steps.
+1. Install [`ember-cli`](https://cli.emberjs.com/release/) globally.
+1. Clone this repo.
+1. `cd ember-component-patterns` and `npm install` dependencies.
+1. Run `ember s` from within the repo to serve the project locally.
```
[the remark this answers is not in the document]
Comment (mattfrankjames) [reply CONTRIBUTING.md:13]: Won't contributors not be able to run any `ember` commands without `ember-cli` installed, though?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+## Guidelines for contribution
+### JS based PRs
+- Contributions to this repo must follot the [Ember Octane](https://emberjs.com/editions/octane) paradigm.
```
[the remark this answers is not in the document]
Comment (mattfrankjames) [reply CONTRIBUTING.md]: ✅

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MelSumner) [reply CONTRIBUTING.md:13]: If they have it installed for the project they would be fine. Folks who use Ember regularly would of course either already have it installed or would know to install it globally. Does that make sense?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+1. Clone this repo.
+1. `cd ember-component-patterns` and `npm install` dependencies.
+1. Run `ember s` from within the repo to serve the project locally.
+
+## Guidelines for contribution
+### JS based PRs
+- Contributions to this repo must follow the [Ember Octane](https://emberjs.com/editions/octane) paradigm.
+- Each new file - whether a route, component, template or controller - should be generated using the `ember g` command.
+- New functionality should be accompanied by a useful test built off the test generated by Ember. Run `npm test` from within the repo to run the test suite locally.
+- JS files must pass the accompanying eslint rules. Run `npm run lint:js` from within the repo to check linting locally.
```
Comment (MelSumner) [inline CONTRIBUTING.md:20]: It would probably be good to include template linting here too. What do you think?

**Review by MelSumner — CHANGES_REQUESTED** [review]
Just a couple of minor changes and we can ship this as a MVP & iterate from there. Thank you for taking care of this!

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mattfrankjames) [reply CONTRIBUTING.md:13]: Yes, that makes sense. I thought it had to be installed globally, but now I see it in the package.json.

File: CONTRIBUTING.md (hunk 3 above)
Comment (mattfrankjames) [reply CONTRIBUTING.md:20]: Agreed, I'll make that update and change the command to npm run lint.

Comment (mattfrankjames) [reply CONTRIBUTING.md:20]: Actually, I'll add a section for template based contributions.

**Review by MelSumner — APPROVED** [review]
Thank you!

**Pull request merged**